### PR TITLE
Story/hfi flexible prep start/1671

### PR DESCRIPTION
**Pull request by conbrad**
- Adds `PrepDateRangeSelector` for selecting prep cycle date ranges
- Encapsulate `start_date`, `end_date` in `DateRange`
- Makes start date inclusive, end date exclusive so we don't get an extra day
- [x] Given I've navigated to the HFI Calc Page, Then there is no longer a "prep days" dropdown (i.e. current functionality to select duration of prep is gone).
- [x] Given I've navigated to the HFI Calc Page, Then the most recently persisted prep period is populated. (see additional context - there will ALWAYS be a prep period saved)
- [x] Given I've changed the prep period, Then pressing the save button will persist the prep period, for the selected fire centre for all users.
- [x] Given I'm using the date range picker, I am limited to a prep period between 1 and 7 days inclusive.
- [ ] Given I've selected the date range picker, I am only able to select a valid prep period.
    - Unable to do this; the component doesn't expose click handlers for each day, only when both days are selected

# Test Links:
[Percentile Calculator](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1810.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: api/app/hfi/hfi_calc.py [hunk 1]
```diff
@@ -156,36 +156,17 @@ async def hydrate_fire_centres():
...
-        end_date = start_date + timedelta(days=2)
-    return start_date, end_date
-
-
 async def calculate_latest_hfi_results(request: HFIResultRequest):
     "Set up time range and fire centre data for calculating HFI results"
 
     # pylint: disable=too-many-locals
-
+    logger.info(request)
```
Comment (Sybrand) [inline api/app/hfi/hfi_calc.py:163]: Do we need to log this?

**Review by Sybrand — COMMENTED** [review]
Some questions & comments.

File: .github/workflows/pr_description.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ on:
 jobs:
   set-description:
     name: Set pull request description
+    permissions: write-all
```
[the remark this answers is not in the document]
Comment (conbrad) [reply .github/workflows/pr_description.yml:11]: Accident, I thought I pushed the revert for this commit.

File: api/app/hfi/hfi_calc.py [hunk 3]
```diff
@@ -347,3 +328,29 @@ def validate_station_daily(daily: StationDaily):
...
+    end_date = date_range.end_date if date_range is not None else None
+    if start_date is None:
+        now = get_pst_now()
+        start_date = date(year=now.year, month=now.month, day=now.day)
+    # don't have an end date, default to start date + 5 days.
+    if end_date is None:
+        end_date = start_date + timedelta(days=4)
+    # check if the span exceeds 7, if it does clamp it down to 7 days.
+    delta = end_date - start_date
+    if delta.days > 6:
```
[the remark this answers is not in the document]
Comment (conbrad) [reply api/app/hfi/hfi_calc.py:351]: It's explained the in the function comment. It's 7 days inclusive of start date, not inclusive of end date.

File: web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx [hunk 4]
```diff
@@ -139,38 +105,30 @@ const HfiCalculatorPage: React.FunctionComponent = () => {
...
-      dispatch(setSaved(false))
       dispatch(
         fetchHFIResult({
           selected_station_code_ids: result.selected_station_code_ids,
           selected_fire_center_id: result.selected_fire_center_id,
           planning_area_fire_starts: result.planning_area_fire_starts,
-          start_date: start.toISODate(),
-          end_date: end.toISODate()
+          date_range: {
+            start_date: newDateRange.startDate?.toISOString().split('T')[0],
```
[the remark this answers is not in the document]
Comment (conbrad) [reply web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx:126]: A js Date doesn't have a timestamp, a Luxon DateTime does

File: api/app/hfi/hfi_calc.py (hunk 1 above)
Comment (conbrad) [reply api/app/hfi/hfi_calc.py:163]: Probably not I can remove

File: api/app/hfi/hfi_calc.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (conbrad) [reply api/app/hfi/hfi_calc.py:351]: I can remove/edit these comments too?

File: web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (conbrad) [reply web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx:126]: Oops I mean `toISODate` doesn't exist on a JS Date, and the timestamp is included in `toISOString` on it

File: api/app/hfi/hfi_calc.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply api/app/hfi/hfi_calc.py:351]: I think so - or at least remove the numbers from the comments - that's what's confusing.

File: web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx:126]: Ok - didn't realize it was js

File: api/app/hfi/hfi_calc.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (conbrad) [reply api/app/hfi/hfi_calc.py:350]: Edited the comments

File: api/app/hfi/hfi_calc.py (hunk 1 above)
Comment (conbrad) [reply api/app/hfi/hfi_calc.py]: Removed

**Review by Sybrand — APPROVED** [review]
Approved assuming comments are addressed and it's been testing at different times of the day to catch that pesky date change when timezones aren't handled correctly.

**Review by tessmgood — APPROVED** [review]
Created follow-up #1815 
Assuming 1634 will impact some of the persist/save AC so didn't look too closely at those ones, fair?

**Review by JulianForeman — APPROVED** [review]

**Review by andrea-williams — APPROVED** [review]

**Pull request merged**